frontend/src/pages/Blogs.tsx

```diff
@@ -15,33 +15,68 @@ export interface BlogType {
   claps: [];
   bookmarkId?: string;
 }
-
 const Blogs = () => {
   const { blogs, loading } = useBlogs();
 
+  const [searchQuery, setSearchQuery] = useState("");
+  const [searchedData, setSearchedData] = useState([]);
+
+  // Search-Logic
+  useEffect(() => {
+    if (searchQuery) {
+      const debouncedSearch = setTimeout(() => {
+        setSearchedData(
+          blogs.filter((info: any) => {
+            return Object.values(info.title)
+              .join('')
+              .toLowerCase()
+              .includes(searchQuery.toLowerCase());
+          })
+        )
+
+        return () => clearTimeout(debouncedSearch);
+      }, 2000);
+    }
+    else {
+      setSearchedData(blogs);
+    }
+  }, [searchQuery])
+
+
   return (
     <>
-      <Appbar skipAuthCheck />
+      <Appbar skipAuthCheck searchQuery={searchQuery} setSearchQuery={setSearchQuery} />
```

Comment:
Once `useSearch` is implemented, there won't be any need to pass searchQuery & setSearchQuery as props.

Appbar can directly get these from the useSearch hook.